packages/headless/src/features/folding/folding-actions.ts

```diff
@@ -67,18 +69,20 @@ export type StateNeededByLoadCollection = ConfigurationSection &
 
 export const loadCollection = createAsyncThunk<
   LoadCollectionFulfilledReturn,
-  CollectionId,
+  ResultWithFolding,
   AsyncThunkSearchOptions<StateNeededByLoadCollection>
 >(
   'folding/loadCollection',
   async (
-    collectionId: CollectionId,
+    result: ResultWithFolding,
```

Comment:
Instead of taking a `result` as a parameter, I think resolving the result as the following might work:
```
const result = state.folding.collections[collectionId]!.result;
```